src/arraytraits.rs

```diff
@@ -94,6 +94,62 @@ impl<S, D, I> IndexMut<I> for ArrayBase<S, D>
     }
 }
 
+impl<'a, A, D> ArrayView<'a, A, D>
+where
+    D: Dimension,
+{
+    /// Get a reference of a element through the view.
+    ///
+    /// This is a replacement of `Index::index` since it forces us a wrong lifetime
+    /// https://github.com/bluss/rust-ndarray/issues/371
+    pub fn elem<I: NdIndex<D>>(&self, index: I) -> &'a A {
+        debug_bounds_check!(self, index);
+        unsafe {
+            &*self.as_ptr().offset(
+                index
+                    .index_checked(&self.dim, &self.strides)
+                    .unwrap_or_else(|| array_out_of_bounds()),
+            )
+        }
+    }
+
+    /// Get a reference of a element through the view without boundary check
+    ///
+    /// This is a replacement of `Index::index` since it forces us a wrong lifetime
+    /// https://github.com/bluss/rust-ndarray/issues/371
+    pub fn uelem<I: NdIndex<D>>(&self, index: I) -> &'a A {
```

Comment:
This method must be `unsafe`. Sorry, I should not have merged without this fixed. I'll fix it now, though.